### ISSUE TITLE
Rule proposal: no same level suites and tests

### ISSUE THREAD

**Issue by fleg** [lo1tuma/eslint-plugin-mocha#104]
Bad
```js
describe('root', function() {
	it('a', function() {});
	it('b', function() {});
	describe('bar', function() {
		it('x', function() {});
		it('y', function() {});
	});
});
```

Good
```js
describe('root', function() {
	describe('foo', function() {
		it('a', function() {});
		it('b', function() {});
	});

	describe('bar', function() {
		it('x', function() {});
		it('y', function() {});
	});
});
```

**Comment by lo1tuma** [comment]
@fleg What would be the rationale for such a rule? Is it purely stylistic or are there any problems with having suites and tests at the same level?

**Comment by fleg** [comment]
stylistic I guess

**Comment by fleg** [comment]
actually mocha do not preserve order for same level suites and tests
```js
describe('root', function() {
    it('a', function() {});
    it('b', function() {});
    describe('bar', function() {
        it('x', function() {});
        it('y', function() {});
    });
    it('q', function() {});
    it('w', function() {});
});
```
```
  root
    ✓ a
    ✓ b
    ✓ q
    ✓ w
    bar
      ✓ x
      ✓ y


  6 passing (16ms)

```

**Comment by lo1tuma** [comment]
Regarding the order of the output of mochas spec reporter, there is already #35 which is a proposal to enforce the same order in the test code.

Anyway I would be ok with I rule that forbids to have tests at the same level as suite. The only thing I wonder if this should be a separate rule than the one suggested in #35 or if they have enough things in common that it could be one generic rule?

**Comment by fleg** [comment]
I think it could be one rule with option to forbid same level tests/suites
